rhombus-lib/rhombus/private/amalgam/rx_match.rhm

```diff
@@ -0,0 +1,25 @@
+#lang rhombus/private/amalgam/core
+import:
+  "when_unless.rhm" open
+
+export:
+  RXMatch
+
+class RXMatch(whole, captures :: List, capture_names :: Map):
+  implements Listable
+  private implements Indexable
```

Comment:
Can we switch this implementation to public and put an annotation like `NonnegInt || Symbol` on `index`?  In general, I think `Indexable` should be publicly implemented so that the `.get` method can be properly documented.

Reply:
I agree about making the implementation public, and an annotation on `index` also sounds good.

Reply:
I made the `indexable` implementation public. I didn't add the annotation after all, because I'm not sure `capture_names` should be restricted to symbol keys. If it is, then it makes sense to use `Map.of` for `capture_names`, but that will add an extra cost to constructing an `RXMatch`. I'm undecided on whether this is a place to be more precise and have more checks versus a place to leave flexible with few checks.